core/modules/qdisp/Executive.cc

```diff
@@ -504,14 +524,16 @@ void Executive::_reapRequesters(std::unique_lock<std::mutex> const&) {
     }
 }
 
-void Executive::_reportStatuses() {
+void Executive::_updateProxyMessages() {
+	{
     std::lock_guard<std::mutex> lock(_statusesMutex);
-    StatusMap::const_iterator i,e;
-    for(i=_statuses.begin(), e=_statuses.end(); i != e; ++i) {
-        ExecStatus::Info info = i->second->getInfo();
+    for(auto i=_statuses.begin(), e=_statuses.end(); i != e; ++i) {
+        JobStatus::Info info = i->second->getInfo();
```

Comment:
`JobStatus::Info const& info`? Or `auto const& info` if you started playing with auto already